src/com/selftest/trafficlight/TrafficLightSimulator.java

```diff
@@ -0,0 +1,62 @@
+package com.selftest.trafficlight;
+
+public class TrafficLightSimulator implements Runnable {
+
+	private Thread t;
+	private TrafficLightColour tlc;
+	boolean stop = false;
+	boolean changed = false;
+
+	public TrafficLightSimulator(TrafficLightColour init) {
+		tlc = init;
+
+		t = new Thread(this);
+		t.start();
+	}
+
+	public TrafficLightSimulator() {
+		this(TrafficLightColour.RED); // default to RED
+	}
+
+	@Override
+	public void run() {
+		while (!stop) {
+			try {
+				Thread.sleep(tlc.getSleepMillis());
+			} catch (InterruptedException e) {
+				System.out.println(e);
+			}
+			changeColour();
+		}
+		
+	}
+
+	public synchronized void changeColour() {
+		// Excuses to use ordinals etc
+		tlc = (tlc.ordinal() == 2) ? TrafficLightColour.values()[0]

```

Comment:
True, my modulus looked clunky in part due to the call to length, which would be needed here to future proof it.

The whole thing would need updated to refer to the on-off state of each bulb in the light box

(RED_n_AMBER's a sort of murky orange, isn't it?)